packages/views.vite-plugin/README.md

```diff
@@ -0,0 +1,163 @@
+# @osdk/views-manifest-vite-plugin
+
+This is a vite plugin that will automatically discover `*.config.(j|t)s` files that are imported into entrypoint files and generates a `.palantir/view-config.json` file for Foundry to read from. This allows developers to write out their configuration for custom views in Foundry in a single, type-safe file and not worry about generating a manifest file needed for publishing new versions of the view within Foundry.
+
+## Usage
+
+To use the plugin, add it to the `plugins` list of your [vite configuration](https://vitejs.dev/config/):
+
+```js
+import viewManifestPlugin from "@osdk/views-manifest-vite-plugin";
+import { defineConfig } from "vite";
+
+export default defineConfig({
+  plugins: [viewManifestPlugin()],
+  // Rest of configuration
+});
+```
+
+## Defining configuration
+
+The `@osdk/views-client` package exports a `ViewConfig` type so that you can define a configuration object in the shape that this plugin requires. You can define your config like so:
+
+```js
+// main.config.ts
+export default {
+  rid: "<FILL IN RID>", // This is the RID of the Foundry view this code will publish to
+  parameters: {
+    headerText: {
+      displayName: "Widget title",
+      type: "string",
+    },
+    showWarning: {
+      displayName: "Show warning callout",
+      type: "boolean",
+    },
+    todoItems: {
+      displayName: "Todo items",
+      type: "array",
+      subType: "string",
+    },
+  },
+  events: {
+    updateHeader: {
+      displayName: "Update header",
+      parameterIds: ["headerText"],
+    },
+    updateTodoItems: {
+      displayName: "Update todo items",
+      parameterIds: ["todoItems"],
+    },
+  },
+} as const satisfies ViewConfig;
+```
+
+## Importing the configuration
+
+Import your `*.config.ts` file in any entrypoint JavaScript/TypeScript file so the plugin can pick it up. Entrypoint files are any that are imported from your HTML file. For example, if you have the following setup:
+
+```
+src/
+  main.tsx
+  main.config.ts
+index.html
+```
+
+And in `index.html`, you import `main.tsx`:
```

Comment:
In practice these view repos won't need any HTML files, do you depend on them for the build? They won't get used in the deployed view so I'd want to avoid users thinking they can add things here that effect the deployed view

Reply:
yes, that's how vite works - entrypoints are the HTML files